### ISSUE TITLE
Campaign Manager (Next Generation)

### ISSUE THREAD

**Issue by sluc23** [systopia/de.systopia.campaign#112]
As previously discussed with Systopia staff (@bjendres , @jensschuppe , @jofranz ) we will take the lead on creating a new branded Campaign Manager extension, using latest coding standards, and removing all the legacy code that is slowly generating issues on newer CiviCRM releases.
The main idea is to keep all the current features, improving the base technology for them, and adopting best practices and modern CiviCRM programming techniques.
We have some new features in mind as well that we'd like to add

We created a new repo for this where initially will place the developments until we decide what's the best path (upgrading this repo, moving to lab.c.o, etc..): https://github.com/ixiam/campaignmanager

We will be working on this repo, adding issues to plan the path of the project and **everyone is welcome to collaborate!**
Thanks

**Comment by jensschuppe** [comment]
Great news, @sluc23! If we are to merge both extensions eventually, bear in mind that their names (aka *extension key*) should either match or we would have to provide an upgrade path from the `campaign` extension to your new `campaignmanager` extension (which would also be totally fine of course, but might require additional effort).

**Comment by sluc23** [comment]
yes, I have that in mind.. I hope the transition won't be too complicated, but I think it's time to change extension namespace from `campaign` to something different, because `campaign` it's too generic and it could lead to additional issues in the future..

**Comment by sluc23** [comment]
Campaign Manager (Next Generation) has been published in [ixiam/campaignmanager](https://lab.civicrm.org/extensions/campaignmanager).
Still a long way to go to add more features, but this first release has some important ones as Campaing Tree generation, KPIs, and Campaign Status Rules.

Thanks

**Comment by ghost** [comment]
Is this extension now published as https://lab.civicrm.org/extensions/campaignmanager/ ?

**Comment by sluc23** [comment]
@eptbertram yes that's the url, but it's a different extension, there's no upgrade path to move from one to the other..

**Comment by jensschuppe** [comment]
As you presented it during the *CiviCamp*, @sluc23, it can't do expenses (you could use the [*Expenses* extension](https://civicrm.org/extensions/expenses), which still seems to be in need of a bit of refactoring for campaigns, iirc), and for visualizing KPIs, I guess the plan was to use the [*ChartKit* extension](https://lab.civicrm.org/ufundo/chart_kit).

I think we'll let this issue open for a while so people can find those information.

**Comment by ghost** [comment]
Great, thx for the fast reply. And for the rewrite of this extension, of course!

I had to disable CiviCampaign module to uninstall campaign-extension and install new compaignmanager-extension. After this, the menu item "Campaign Dashboard" was missing (URL /civicrm/campaign). I'm testing right now ;)

**Comment by GFF-DS** [comment]
I am confused which extension to use. Will de.systopia.campaign become deprecated in the future in favor of campaignmanager?

**Comment by jensschuppe** [comment]
@GFF-DS That's the plan eventually, but there's no feature parity yet AFAIK (regarding KPI vizualisations and expenses), and also no upgrade path from one ext to the other. We fix bugs in *de.systopia.campaign* though when there's funding or PRs, but we don't actively maintain or provide support for it anymore.

When deciding which extension to start with, I'd say use ixiam's *campaignmanager*. If you're already using *de.systopia.campaign* and are happy with its current features, stick with it.

**Comment by GFF-DS** [comment]
Thanks for the fast answer!
